source/wiz/command_line.py

```diff
@@ -1966,3 +1971,75 @@ def _export_history_if_requested(click_context):
     )
     wiz.filesystem.export(path, history, compressed=True)
     logger.info("History recorded and exported in '{}'".format(path))
+
+
+def _compare_semver(left, right):
+    """
+    Compares two semantic versions as per https://semver.org/ with a small
+    adjustment. The regex allows for missing patch versions as well as custom
+    pre-release tags. Intended to be used with Python's `sorted` function.
+
+    :param left: First semantic version to compare
+    :type left: String
+    :param right: Second semantic version to compare
+    :type right: String
+
+    :return: Returns 0 if versions are equal, -1 if left < right,
+             or 1 if left > right
+    """
+    parse_semver = re.compile(
```

Comment:
It's not the official semver regex. It's edited to allow versions like `1.0.4b0` like some projects I've seen use

Reply:
Oh but this regex should cover it as well:
```python
>>> from packaging.version import VERSION_PATTERN
>>> match = re.search(VERSION_PATTERN, "1.0.4b0", re.VERBOSE).group("release")
'1.0.4'
```

And actually, thinking about this more, I feel you might actually be able to simplify the logic greatly by using [packaging.version](https://packaging.pypa.io/en/latest/version/) to handle the comparison:
```python
>>> Version('4.21.4') > Version('1.0.0-rc.1') 
True

>>> Version('4.21.4') > Version('11') 
False
```

The only edge case you'd have to handle is if the definition has not version, which should theoretically never happen:
```python
from packaging.version import Version, InvalidVersion

def compare_version(version1, version2):
    try:
        version1 = Version(version1)
    except InvalidVersion:
        pass

    try:
        version2 = Version(version2)
    except InvalidVersion:
        pass

    if type(version1) == type(version2):
        if identifier1 == identifier2:
            return 0
        return -1 if version1 < version2 else 1

    elif isinstance(version1, Version):
        return -1
    return 1
```

I actually used this method in Qip when comparing variants:
https://github.com/themill/qip/blob/master/source/qip/definition.py#L352

Sorry, I should have pointing that out on the first review...

Reply:
That does look much cleaner. I'll have a look at using that. Hopefully it can handle more than the semver regex. Would make the comparison function a heck of a lot simpler.

Reply:
That doesn't support the `1.0.0-alpha.beta` format, or more generically the semantic version defined _<dot-separated pre-release identifiers>_  https://semver.org/#backusnaur-form-grammar-for-valid-semver-versions

If that's ok, then I'm happy to use the packaging version comparison. Let me know

Reply:
Sorry for the very late answer, I'd prefer sticking to the logic defined in [PEP 440](https://www.python.org/dev/peps/pep-0440) as it is the one used everywhere else to handle versioning. The `1.0.0-alpha.beta` format would not be allowed at the moment so it won't be an issue:

```python
>>> from packaging.version import Version
>>> Version("1.0.0-alpha.beta")
...
InvalidVersion: Invalid version: '1.0.0-alpha.beta'
```

Saying that, I feel that it would be worth revisiting the logic handling versioning and requirement specifier at some point to use pure semantic versioning, but it is a bigger task as it requires to get rid of the dependency to the [packaging](https://packaging.pypa.io/en/latest/) module. 

I added a separated issue to handle this: #64

Reply:
Just pushed the modification, if you're fine with that I'll merge it tomorrow!

Reply:
Code's a little confusing, but LGTM

Reply:
Compromise found after a call :)